soccer/gameplay/plays/offense/distraction1.py

```diff
@@ -0,0 +1,188 @@
+                        behavior.Behavior.State.running)
+        self.add_state(distraction1.State.shoot,
+                        behavior.Behavior.State.running)
+        
+        
+        self.add_transition(behavior.Behavior.State.start,
+                        distraction1.State.setup, lambda: True, 'immediately')
+
+        self.add_transition(distraction1.State.setup,
+                        distraction1.State.passing, lambda: not self.ball_is_far,'1-passing')
+
+        #optionaladjustment can be skipped if ball is not far from goal
+        self.add_transition(distraction1.State.setup, 
+                        distraction1.State.optionaladjustment, lambda: self.ball_is_far and 
+                        self.subbehavior_with_name('striker moves').is_done_running(),'1-2')
+
+        self.add_transition(distraction1.State.optionaladjustment,
+                        distraction1.State.centerpass, lambda: self.subbehavior_with_name('move half').is_done_running() and 
+                        self.subbehavior_with_name('move to distract').is_done_running() , '2-centerpass')
+
+        self.add_transition(distraction1.State.centerpass,
+                        distraction1.State.passing, lambda: self.subbehavior_with_name('center pass').is_done_running(), 'centerpass-passing')
+
+        self.add_transition(distraction1.State.passing, 
+                        distraction1.State.cross, lambda: (self.has_subbehavior_with_name('distract pass') and self.subbehavior_with_name('distract pass').is_done_running() )
+                        , 'passing-crossing')
+
+        self.add_transition(distraction1.State.passing,
+                        distraction1.State.shoot, lambda: (self.has_subbehavior_with_name('striker pass') and self.subbehavior_with_name('striker pass').is_done_running() ),'pass-shoot' )
+
+        self.add_transition(distraction1.State.cross,
+                        distraction1.State.shoot, lambda: self.subbehavior_with_name('pass to striker').is_done_running()
+                        , 'crossing-shooting')
+
+        self.add_transition(distraction1.State.shoot,
+                        distraction1.State.setup, lambda: self.has_subbehavior_with_name('shooting') and self.subbehavior_with_name('shooting').is_done_running(), 'repeat')
+
+        
+        self.d1 = robocup.Point(0.40*constants.Field.Width, 0.95*constants.Field.Length) #the first distraction point
+        self.d2 = robocup.Point(0.40*constants.Field.Width, 0.8*constants.Field.Length) #the second distraction point
+        self.s1 = robocup.Point(-0.40*constants.Field.Width, 0.9*constants.Field.Length) #striker's position
+        self.center = robocup.Point(0,4.5) #center of field position, used if ball is far
+        
+
+    def on_enter_setup(self):
+        #capture ball and get striker and distractor in position
+        print("entered setup")
+        self.add_subbehavior(skills.capture.Capture(), 'capture1')
```

Comment:
All `add_subbehavior` calls should have a `required` parameter. See [here](https://github.com/RoboJackets/robocup-software/blob/master/soccer/gameplay/plays/offense/adaptive_formation.py#L243). And if it's a subbehavior that doesn't have to happen, add a lower priority to it. The distraction robot is a prime example

Reply:
10 is a good number for non-essential roles